psl-core/src/test/java/org/linqs/psl/evaluation/statistics/RankingEvaluatorTest.java

```diff
@@ -29,6 +43,77 @@ protected RankingEvaluator getComputer() {
+        Constant[][] cannedTerms = new Constant[nCannedTerms][];
+        cannedTerms[0] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(1) };
+        cannedTerms[1] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(3) };
+        cannedTerms[2] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(4) };
+        cannedTerms[3] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(1) };
+        cannedTerms[4] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(2) };
+        cannedTerms[5] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(3) };
+        cannedTerms[6] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(4) };
+        cannedTerms[7] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(2) };
+
+        // Insert the predicated values.
+        Inserter inserter = dataStore.getInserter(predicate, targetPartition);
+        double predValue = 1.0;
+        double step = predValue / (double)nCannedTerms;
+        for (Constant[] terms : cannedTerms) {
+            // Default prediction should be in the order of the cannedTerms
+            // instances above as they all have predValue 1
+            inserter.insertValue(predValue, terms);
+            predValue = predValue - step;
+        }
+
+        // create some ground truth atoms
+        Constant[][] baselinePositiveTerms = new Constant[6][];
+        baselinePositiveTerms[0] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(1) };
+        baselinePositiveTerms[1] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(3) };
+        baselinePositiveTerms[2] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(4) };
+        baselinePositiveTerms[3] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(1) };
+        baselinePositiveTerms[4] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(2) };
+        baselinePositiveTerms[5] = new Constant[]{ new UniqueIntID(2), new UniqueIntID(3) };
+
+        Constant[][] baselineNegativeTerms = new Constant[1][];
+        baselineNegativeTerms[0] = new Constant[]{ new UniqueIntID(1), new UniqueIntID(2) };
+
+        // Insert the truth values.
+        inserter = dataStore.getInserter(predicate, truthPartition);
+        for (Constant[] terms : baselinePositiveTerms) {
+            inserter.insertValue(1.0, terms);
+        }
+        for (Constant[] terms : baselineNegativeTerms) {
+            inserter.insertValue(0.0, terms);
+        }
+
+        // Redefine the truth database with no atoms in the write partition.
+        Database results = dataStore.getDatabase(targetPartition);
+        Database truth = dataStore.getDatabase(truthPartition, dataStore.getRegisteredPredicates());
+
+        PersistedAtomManager atomManager = new PersistedAtomManager(results);
+        trainingMap = new TrainingMap(atomManager, truth, true, false);
+
+        // Since we only need the map, we can close all the databases.
+        results.close();
+        truth.close();
+    }
+
     @Test
     public void testAUROC() {
```

Comment:
Add a test(s) that captures your new trapezoid change.
(The existing code should fail (by not giving a precise enough answer), while your code should pass.)